sql/core/src/test/scala/org/apache/spark/sql/execution/command/ShowTablesSuite.scala

```diff
@@ -0,0 +1,113 @@
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{BooleanType, StringType, StructType}
+
+trait ShowTablesSuite extends QueryTest with SharedSparkSession {
+  protected def catalog: String
+  protected def defaultUsing: String
+  case class ShowRow(namespace: String, table: String, isTemporary: Boolean)
+  protected def getRows(showRows: Seq[ShowRow]): Seq[Row]
+  // Gets the schema of `SHOW TABLES`
+  protected def showSchema: StructType
+
+  protected def runShowTablesSql(sqlText: String, expected: Seq[ShowRow]): Unit = {
+    val df = spark.sql(sqlText)
+    assert(df.schema === showSchema)
+    assert(df.collect() === getRows(expected))
+  }
+
+  protected def withSourceViews(f: => Unit): Unit = {
+    withTable("source", "source2") {
+      val df = spark.createDataFrame(Seq((1L, "a"), (2L, "b"), (3L, "c"))).toDF("id", "data")
+      df.createOrReplaceTempView("source")
+      val df2 = spark.createDataFrame(Seq((4L, "d"), (5L, "e"), (6L, "f"))).toDF("id", "data")
+      df2.createOrReplaceTempView("source2")
+      f
+    }
+  }
+
+  test("show an existing table") {
+    val namespace = "test"
+    val table = "people"
+    withDatabase(s"$catalog.$namespace") {
+      sql(s"CREATE DATABASE $catalog.$namespace")
+      withTable(s"$catalog.$namespace.$table") {
+        sql(s"CREATE TABLE $catalog.$namespace.$table (name STRING, id INT) $defaultUsing")
+        runShowTablesSql(s"SHOW TABLES IN $catalog.test", Seq(ShowRow(namespace, table, false)))
+      }
+    }
+  }
+
+  test("show tables with a pattern") {
+    withDatabase(s"$catalog.db", s"$catalog.db2") {
+      sql(s"CREATE DATABASE $catalog.db")
+      sql(s"CREATE DATABASE $catalog.db2")
+      withTable(
+        s"$catalog.db.table",
+        s"$catalog.db.table_name_1",
+        s"$catalog.db.table_name_2",
+        s"$catalog.db2.table_name_2") {
+        sql(s"CREATE TABLE $catalog.db.table (id bigint, data string) $defaultUsing")
+        sql(s"CREATE TABLE $catalog.db.table_name_1 (id bigint, data string) $defaultUsing")
+        sql(s"CREATE TABLE $catalog.db.table_name_2 (id bigint, data string) $defaultUsing")
+        sql(s"CREATE TABLE $catalog.db2.table_name_2 (id bigint, data string) $defaultUsing")
+
+        runShowTablesSql(
+          s"SHOW TABLES FROM $catalog.db",
+          Seq(
+            ShowRow("db", "table", false),
+            ShowRow("db", "table_name_1", false),
+            ShowRow("db", "table_name_2", false)))
+
+        runShowTablesSql(
+          s"SHOW TABLES FROM $catalog.db LIKE '*name*'",
+          Seq(
+            ShowRow("db", "table_name_1", false),
+            ShowRow("db", "table_name_2", false)))
+
+        runShowTablesSql(
+          s"SHOW TABLES FROM $catalog.db LIKE '*2'",
+          Seq(ShowRow("db", "table_name_2", false)))
+      }
+    }
+  }
+
+  // TODO(SPARK-33393): Support SHOW TABLE EXTENDED in DSv2
+  test("SHOW TABLE EXTENDED for default") {
```

Comment:
this test assumes that the default catalog is v1 catalog. Shall we move it to `v1.ShowTablesSuite`?

Reply:
DSv2 doesn't support `SHOW TABLE EXTENDED`. After https://issues.apache.org/jira/browse/SPARK-33393, the test can work for DSv2 too with minimal changes:
- Abstract the column name `database` - in DSv2 we should show `namespace` probably
- DSv2 will not show views (at least at the moment).

I would split the test to 2 tests:
- SHOW TABLE EXTENDED for tables
- SHOW TABLE EXTENDED for views.

Reply:
My point is that: the tests in the base trait, should work for both v1 and v2 catalogs, and will be run with both v1 and v2 catalog.

The test now just run twice repeatedly in `v1.ShowTablesSuite` and `v2.ShowTablesSuite`, which doesn't make sense.

Reply:
Create 2 separate tests for dsv1 and dsv2